File: bioskopina_mobile/lib/models/donation.dart
import 'package:json_annotation/json_annotation.dart';
import '../models/user.dart';
part 'donation.g.dart';

@JsonSerializable()
class Donation {
  int? id;
  int? userId;
  double? amount;
  DateTime? dateDonated;
  String? transactionId;
  User? user;

  Donation({
    this.id,
    this.userId,
    this.amount,
    this.dateDonated,
    this.transactionId,
    this.user,
  });

  factory Donation.fromJson(Map<String, dynamic> json) =>
      _$DonationFromJson(json);

  Map<String, dynamic> toJson() => _$DonationToJson(this);
}

File: backend/Bioskopina.Services/Database/Donation.cs
using System;
using System.Collections.Generic;

namespace Bioskopina.Services.Database;

public partial class Donation
{
    public int Id { get; set; }

    public int UserId { get; set; }

    public decimal Amount { get; set; }

    public DateTime DateDonated { get; set; }

    public string TransactionId { get; set; } = null!;

    public virtual User User { get; set; } = null!;
}
